x-pack/legacy/plugins/maps/public/layers/vector_layer.js

```diff
@@ -702,6 +745,7 @@ export class VectorLayer extends AbstractLayer {
   ownsMbLayerId(mbLayerId) {
     return (
       this._getMbPointLayerId() === mbLayerId ||
+      this._getMbTextLayerId() === mbLayerId ||
```

Comment:
for symmetry, add the text-layer-id here as well 

https://github.com/thomasneirynck/kibana/blob/e669be7e9240831068ddcd1de25885714d390256/x-pack/legacy/plugins/maps/public/layers/vector_layer.js#L737-L744